README.md

```diff
@@ -157,136 +139,121 @@ symbols, it will fail at class definition time. This program will throw:
 ```js
 protocol I { a; b; }
 
-class C implements I {
-  [I.a]() {}
-  // note the missing implementation of I.b
-}
-```
-
-### minimal implementations
-
-Minimal implementations can be expressed using protocol inheritance.
-
-```js
-// Applicative elided
-protocol Monad extends Applicative {
-  bind;
-  join;
-  kleisli() {}
-}
-
-// two possible minimal implementations for Monad
-protocol MonadViaBind extends Monad {
-  [Monad.join]() { /* default implementation in terms of bind elided */ }
-}
-protocol MonadViaJoin extends Monad {
-  [Monad.bind]() { /* default implementation in terms of join elided */ }
-}
-
-class C implements MonadViaBind {
-  [Monad.bind]() {}
-}
-class D implements MonadViaJoin {
-  [Monad.join]() {}
+class C {
+  implements I {
+    a() {}
+    // b() {}
+    // ^__ throws PartialImplError
+  }
 }
 ```
 
-### `Protocol.implement`
+### `implFor <class>`
 
 An important aspect of this proposal is that it needs to be possible to apply
 a protocol to an existing class.
 
 ```js
 protocol Functor {
   map;
-}
 
-Array.prototype[Functor.map] = Array.prototype.map;
-Protocol.implement(Array, Functor);
+  implFor Array {
```

Comment:
This doesn’t seem to conflict with a symbol named implFor as long as there’s a NLT. If not, why reserve a symbol name? If so, why not use something that can’t be an identifier name, is that we don’t need to arbitrarily reserve a name or use an NLT?

I’m thinking of either using a keyword like implements or interface, or, some kind of sigil?

Reply:
Separately, does this implicitly access .prototype of the thing i provide? Does it have to be a constructor or even a function, or I’d it just sugar for ToObject().prototype? How would i install methods on Array itself (not Array.prototype)?

Reply:
There's no syntax conflict in the body because we have full control of the internals of protocol. `implements` inside class bodies is more of a concern and that one would need an `async`-style NLTH, but the braces can go in the next line just fine (since it's only one non-whitespace token of lookahead?)